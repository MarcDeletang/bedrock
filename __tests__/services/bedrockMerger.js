const importApp = require('../../app');

const { BedrockConfigMergerService } = importApp;

const promise = () => Promise.resolve(true);

const services = [];
const models = [];
const registers = [];
const decorations = [];
const controllers = {};
const hapiEvents = [];

const services2 = [{ init: promise, foo: 'bar' }];
const models2 = [{ init: promise }];
const registers2 = [promise];
const decorations2 = [{ target: 'foo', name: 'bar', method: promise }];
const controllers2 = {
  foo: {
    PATH: 'HELLO',
    handler: () => true,
  }
};

const hapiEvents2 = [
  { name: 'onPreAuth', method: ()=> true }
];

describe('Testing bedrock config merger', () => {

  it('Check the service shape', () => {
    expect(BedrockConfigMergerService).toMatchSnapshot();
  });

  it('Testing merge empty params', () => {
    const config1 = { services, models, registers, decorations, controllers, hapiEvents };
    const config2 = { services, models, registers, decorations, controllers, hapiEvents };
    expect(BedrockConfigMergerService.merge(config1, config2)).toMatchSnapshot();
  });

  it('Testing merge empty and not empty', () => {
    const config1 = { services, models, registers, decorations, controllers, hapiEvents };
    const config2 = {
      services: services2,
      models: models2,
      registers: registers2,
      decorations: decorations2,
      controllers: controllers2,
      hapiEvents: hapiEvents2,
    };
    expect(BedrockConfigMergerService.merge(config1, config2)).toMatchSnapshot();
  });

  it('Testing merge not empty and not empty', () => {
    const config1 = {
      services: services2,
      models: models2,
      registers: registers2,
      decorations: decorations2,
      controllers: controllers2,
      hapiEvents: hapiEvents2,
    };
    const config2 = {
      services: services2,
      models: models2,
      registers: registers2,
      decorations: decorations2,
      controllers: controllers2,
      hapiEvents: hapiEvents2,
    };
    expect(BedrockConfigMergerService.merge(config1, config2)).toMatchSnapshot();
  });

});
