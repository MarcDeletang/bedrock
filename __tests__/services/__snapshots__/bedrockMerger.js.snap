// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing bedrock config merger Check the service shape 1`] = `
Object {
  "merge": [Function],
}
`;

exports[`Testing bedrock config merger Testing merge empty and not empty 1`] = `
Object {
  "controllers": Object {
    "0": Object {},
    "1": Object {
      "foo": Object {
        "PATH": "HELLO",
        "handler": [Function],
      },
    },
  },
  "decorations": Array [
    Object {
      "method": [Function],
      "name": "bar",
      "target": "foo",
    },
  ],
  "hapiEvents": Array [
    Object {
      "method": [Function],
      "name": "onPreAuth",
    },
  ],
  "models": Array [
    Object {
      "init": [Function],
    },
  ],
  "registers": Array [
    [Function],
  ],
  "services": Array [
    Object {
      "foo": "bar",
      "init": [Function],
    },
  ],
}
`;

exports[`Testing bedrock config merger Testing merge empty params 1`] = `
Object {
  "controllers": Object {
    "0": Object {},
    "1": Object {},
  },
  "decorations": Array [],
  "hapiEvents": Array [],
  "models": Array [],
  "registers": Array [],
  "services": Array [],
}
`;

exports[`Testing bedrock config merger Testing merge not empty and not empty 1`] = `
Object {
  "controllers": Object {
    "0": Object {
      "foo": Object {
        "PATH": "HELLO",
        "handler": [Function],
      },
    },
    "1": Object {
      "foo": Object {
        "PATH": "HELLO",
        "handler": [Function],
      },
    },
  },
  "decorations": Array [
    Object {
      "method": [Function],
      "name": "bar",
      "target": "foo",
    },
    Object {
      "method": [Function],
      "name": "bar",
      "target": "foo",
    },
  ],
  "hapiEvents": Array [
    Object {
      "method": [Function],
      "name": "onPreAuth",
    },
    Object {
      "method": [Function],
      "name": "onPreAuth",
    },
  ],
  "models": Array [
    Object {
      "init": [Function],
    },
    Object {
      "init": [Function],
    },
  ],
  "registers": Array [
    [Function],
    [Function],
  ],
  "services": Array [
    Object {
      "foo": "bar",
      "init": [Function],
    },
    Object {
      "foo": "bar",
      "init": [Function],
    },
  ],
}
`;
